packages/ember-glimmer/lib/environment.js

```diff
@@ -252,7 +254,7 @@ export default class Environment extends GlimmerEnvironment {
   // a Compiler can wrap the template so it needs its own cache
   getCompiledBlock(Compiler, template, owner) {
     let compilerCache = this._compilerCache.get(Compiler);
-    return compilerCache.get(template, owner);
+    return compilerCache.get({ template, owner });
```

Comment:
as a follow up we may want to add a context object we reuse so we don't have to keep allocating this object for the lookup but we can follow up on that